src/services/authService.js

```diff
@@ -176,4 +175,34 @@ export default class AuthService {
 
     return 'Verification Link Sent';
   }
+
+  /**
+ * @param {number} userId - ID of logged in user
+ * @return {object} - object containing user profile information
+ */
+  static async getProfile(userId) {
+    const result = await User.findOne({ where: { id: userId } });
+    const { dataValues: user } = result;
+    return Helper.omitFields(user, ['password']);
+  }
+
+  /**
+* @param {number} userId - ID of logged in user
+* @param {object} profileDetails - object containing details to be changed
+* @return {object} - object containing updated user profile information
+*/
+  static async updateProfile(userId, profileDetails) {
+    const importantFields = Helper.pickFields(profileDetails, [
```

Comment:
I didn't know about `.optional()` in `@hapi/joi` then, I'll change that now.

Reply:
Why do you want to use `.optional() in @hapi/joi`?

Reply:
So that the user doesn't necessarily have to enter all the details in the schema if he/she is not updating everything. For example, if the user is just updating gender, he shouldn't have to type everything else as well.

Reply:
You don't need all this because you provide the fields you want the user to update and a user can practically update all his info except database `id,createdAt and updatedAt`

Reply:
Alright